tests/test_views.py

```diff
@@ -1,85 +1,82 @@
-        staff_user = self.get_superuser()
-        with self.login_user_context(staff_user):
-            response = self.client.get(self.preview_url)
-            self.assertEqual(response.status_code, 200)
+
+class PreviewViewTestCases(CMSTestCase):
+    def setUp(self):
+        self.menu_content = MenuContentFactory()
+        self.preview_url = admin_reverse(
+            "djangocms_navigation_menuitem_preview",
+            kwargs={"menu_content_id": self.menu_content.id},
+        )
+
+    def test_view_url(self):
+        expected_url = "/en/admin/djangocms_navigation/menuitem/{}/preview/".format(
+            self.menu_content.id
+        )
+        self.assertEqual(self.preview_url, expected_url)
 
     def test_view_context_data_with_valid_menu_content(self):
-        factory = RequestFactory()
-        request = factory.get(self.preview_url)
-        request.user = self.get_superuser()
-        view = MenuContentPreviewView()
-        view.request = request
-        view.kwargs = {"menu_content_id": self.menu_content.id}
-        response = view.get_context_data()
-        self.assertIsInstance(response["view"], MenuContentPreviewView)
-        self.assertIn("annotated_list", response)
-
-    def test_view_context_data_with_invalid_int_menu_content(self):
-        factory = RequestFactory()
-        request = factory.get(self.preview_url)
-        request.user = self.get_superuser()
-        view = MenuContentPreviewView()
-        view.request = request
-        view.kwargs = {"menu_content_id": 99}
-        response = view.get_context_data()
-        self.assertEqual(response.status_code, 400)
-
-    def test_view_context_data_with_invalid_string_menu_content(self):
-        factory = RequestFactory()
-        request = factory.get(self.preview_url)
-        request.user = self.get_superuser()
-        view = MenuContentPreviewView()
-        view.request = request
-        view.kwargs = {"menu_content_id": "dummy"}
-        response = view.get_context_data()
-        self.assertEqual(response.status_code, 400)
+        super_user = self.get_superuser()
+        with self.login_user_context(super_user):
+            response = self.client.get(self.preview_url)
+            expected = MenuItem.get_annotated_list(parent=self.menu_content.root)
+            self.assertEqual(response.status_code, 200)
+            self.assertIn("annotated_list", response.context_data)
+            self.assertEqual(response.context_data["annotated_list"], expected)
+
+    def test_view_response_with_invalid_int_menu_content(self):
+        super_user = self.get_superuser()
+        with self.login_user_context(super_user):
+            preview_url = admin_reverse(
+                "djangocms_navigation_menuitem_preview", kwargs={"menu_content_id": 99}
+            )
+            response = self.client.get(preview_url)
+            self.assertEqual(response.status_code, 404)
+
+    def test_view_url_with_invalid_string_menu_content(self):
```

Comment:
I do think this test is probably slightly overkill. I mean testing for the non-existence of urls means we could potentially be testing this for any url! It made sense to have a test for ValueError, but not sure it does for this.